history/history-impl/src/main/java/com/duckduckgo/history/impl/RealNavigationHistory.kt

```diff
@@ -17,18 +17,25 @@
 package com.duckduckgo.history.impl
 
 import com.duckduckgo.app.browser.DuckDuckGoUrlDetector
+import com.duckduckgo.common.utils.CurrentTimeProvider
 import com.duckduckgo.di.scopes.AppScope
 import com.duckduckgo.history.api.HistoryEntry
 import com.duckduckgo.history.api.NavigationHistory
 import com.squareup.anvil.annotations.ContributesBinding
 import io.reactivex.Single
 import javax.inject.Inject
 
-@ContributesBinding(AppScope::class)
+interface InternalNavigationHistory : NavigationHistory {
```

Comment:
why we need this one here if you have `HistoryRepository` that seems to be your internal Navigation Repository?

Reply:
`RealNavigationHistory` is what contains the business logic, whereas `NavigationRepository` is a bridge between `RealNavigationHistory` (logic) and the data sources. This internal interface is added so we don't expose externally the ability to schedule data deletion. I don't think it would make sense for the repository to define the time interval for data deletion